app/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/javalin/DomainsRegistrationTest.groovy

```diff
@@ -97,14 +103,166 @@ class DomainsRegistrationTest extends Specification {
+
+        TestApplicationContext.injectRegisteredImplementations()
+
+        when:
+        def res = DomainsRegistration.authenticateRequest(new DomainRequest())
+        def actualStatusCode = res.getStatusCode()
+
+        then:
+        noExceptionThrown()
+        actualStatusCode == expectedStatusCode
+    }
+
+    def "authenticateRequest SecretRetrievalException unhappy path works"() {
+        given:
+        def expectedStatusCode = 500
+
+        def mockValidator = Mock(AuthRequestValidator)
+        mockValidator.isValidAuthenticatedRequest(_ as DomainRequest) >> { throw new SecretRetrievalException("internal error", new IllegalArgumentException()) }
+        TestApplicationContext.register(AuthRequestValidator, mockValidator)
+
+        TestApplicationContext.register(Formatter, Jackson.getInstance())
+
+        TestApplicationContext.injectRegisteredImplementations()
+
+        when:
+        def res = DomainsRegistration.authenticateRequest(new DomainRequest())
+        def actualStatusCode = res.getStatusCode()
+
+        then:
+        actualStatusCode == expectedStatusCode
+    }
+
+    def "authenticateRequest IllegalArgumentException unhappy path works"() {
+        given:
+        def expectedStatusCode = 500
+
+        def mockValidator = Mock(AuthRequestValidator)
+        mockValidator.isValidAuthenticatedRequest(_ as DomainRequest) >> { throw new IllegalArgumentException("internal error", new IllegalArgumentException()) }
+        TestApplicationContext.register(AuthRequestValidator, mockValidator)
+
+        TestApplicationContext.register(Formatter, Jackson.getInstance())
+
+        TestApplicationContext.injectRegisteredImplementations()
+
+        when:
+        def res = DomainsRegistration.authenticateRequest(new DomainRequest())
+        def actualStatusCode = res.getStatusCode()
+
+        then:
+        actualStatusCode == expectedStatusCode
+    }
+
+    def "processRequest happy path works"() {
+        given:
+        def isProtected = true
+        def request = new DomainRequest()
+        def handler =  { DomainRequest req ->
+            return new DomainResponse(200)
+        }
+        def expectedStatusCode = (handler as Function<DomainRequest, DomainResponse>).apply(request).statusCode
+
+        def mockValidator = Mock(AuthRequestValidator)
+        mockValidator.isValidAuthenticatedRequest(_ as DomainRequest) >> true
+        TestApplicationContext.register(AuthRequestValidator, mockValidator)
+
+        TestApplicationContext.injectRegisteredImplementations()
+
+        when:
+        def actualStatusCode = DomainsRegistration.processRequest(request, handler, isProtected).statusCode
+
+        then:
+        actualStatusCode == expectedStatusCode
+    }
+
+    def "processRequest unhappy path works"() {
+        given:
+        def isProtected = true
+        def request = new DomainRequest()
+        def handler =  { DomainRequest req ->
+            return new DomainResponse(401)
```

Comment:
I recommend changing the handler's response status code to something that isn't the same as the response we expect to come from the failed authentication check.  Else, we can't differentiate between us failing the authentication check or calling the handler.